go/cmd/vtbackup/vtbackup.go

```diff
@@ -128,10 +128,12 @@ func registerFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&allowFirstBackup, "allow_first_backup", allowFirstBackup, "Allow this job to take the first backup of an existing shard.")
 	fs.BoolVar(&restartBeforeBackup, "restart_before_backup", restartBeforeBackup, "Perform a mysqld clean/full restart after applying binlogs, but before taking the backup. Only makes sense to work around xtrabackup bugs.")
 	// vttablet-like flags
+	fs.BoolVar(&keepTemporaryFiles, "keep-temporary-files", keepTemporaryFiles, "Retain temporary files and directories after exit.")
 	fs.StringVar(&initDbNameOverride, "init_db_name_override", initDbNameOverride, "(init parameter) override the name of the db used by vttablet")
 	fs.StringVar(&initKeyspace, "init_keyspace", initKeyspace, "(init parameter) keyspace to use for this tablet")
 	fs.StringVar(&initShard, "init_shard", initShard, "(init parameter) shard to use for this tablet")
 	fs.IntVar(&concurrency, "concurrency", concurrency, "(init restore parameter) how many concurrent files to restore at once")
+	fs.StringVar(&tabletPath, "tablet-path", tabletPath, "tablet alias")
```

Comment:
Hey @mattlord I should have documented the intention in the PR description, but we can hash out the details here first, and once we decide whether these flags are needed I can improve the PR description and (if we keep them) flag descriptions.

> Were these flags added for e2e tests?

Yes, the only reason I added these flags was so that I could make the e2e tests work. 

Prior to the PR, the `vtbackup` binary [deletes all temporary files](https://github.com/vitessio/vitess/blob/55695d1/go/cmd/vtbackup/vtbackup.go#L244-L250) associated with the backup, including MySQL data and logs. I wasn't sure how else to answer the question "did MySQL disable and then redo the redo log during its run" without looking at the logs. Thus the `--keep-temporary-files` flag.

Prior to this PR, `vtbackup` internally [invents a random tablet alias](https://github.com/vitessio/vitess/blob/55695d1/go/cmd/vtbackup/vtbackup.go#L232-L239) for each run, and that [alias is then included in the file path](https://github.com/vitessio/vitess/blob/55695d1/go/cmd/vtbackup/vtbackup.go#L244) where log files are stored. I couldn't see how to figure out the log file location from the E2E tests, so I came up with passing `--tablet-path` in as a solution.

Now to answer some of your other questions...

> Is it a full tablet-alias, which determines the sub-directory to use within a backup dir? Or is it just a tablet alias? It doesn't seem like it's a full path, is it?

It is the path to the tablet data, and includes the tablet alias. It could be either a full path or a relative path. Internally, during a normal run `vtbackup` calculates a random relative path, but I believe a full path would work as well.

> So maybe we should just call this flag --tablet-uid or --tablet-alias

That would be fine with me, and would still let me solve some of the problems I ran into during the E2E tests. The alias is the only random part of the path. As long as I can be confident that `vtbackup` will store logs in a deterministic location, I can answer the question "did MySQL disable and then redo the redo log during its run".

FWIW the reason I chose `--tablet-path` was for consistency with other VT binaries. E.g. `vttablet` binary has [a `--tablet-path` flag](https://github.com/vitessio/vitess/blob/55695d1/go/cmd/vttablet/vttablet.go#L87).

> Why not make the flag default be vtbackup instead of hiding that in the code?

The behavior of `vtbackup` prior to this PR is to choose a random tablet alias each run. [There is a documented reason for that](https://github.com/vitessio/vitess/blob/55695d1/go/cmd/vtbackup/vtbackup.go#L227-L231), and I wasn't confident that I should change the logic.

> What's the use case for making this a user variable? I'm not sure how this part relates to the PR

As you sensed towards the end of your comment, these flags are purely to make it easier to write the E2E test.

> I feel like adding this flag only makes things more confusing for users w/o a clear use case it meets

Totally agree. Would prefer not to have these flags if we can think of a better way to answer the "did MySQL disable and then redo the redo log during its run" question. Alternately, happy to adjust the flags or give them better names. Or maybe pflags has a way to "hide" flags, so they don't show up when users run `--help`?

> The tablet-path also seems to impact the cell used. Is there any impact there? Is that part of why we allow the user to specify?

There's no impact to choosing a fake or made-up cell. `vtbackup` [already chooses a made-up cell](https://github.com/vitessio/vitess/blob/55695d1/go/cmd/vtbackup/vtbackup.go#L237). However, I'm not sure what would happen if a user chose a real cell and an already-existing UID. That might be bad.

Being able to choose the cell wasn't the motivation. Only motivation is being able to determine the location of MySQL log files.